dev/build_and_publish.dot

```diff
@@ -0,0 +1,135 @@
+	color=lightgrey;
+
+
+
+    node [style=filled,color=white];
+
+    Trigger -> build_bom
+    bom_built -> build_containers
+    bom_built -> build_debians
+    bom_built -> build_rpms
+    bom_built -> build_changelog
+    bom_built -> build_apidocs
+
+
+    subgraph cluster_0 {
+        label = "Build BOM"
+
+        build_bom -> git_clone_at_branch;
+
+        subgraph cluster_1 {
+		    label = "For Each Service..."
+		    style=filled;
+		    color=aquamarine3;
+
+    		git_clone_at_branch -> get_HEAD_hash;
+    		get_HEAD_hash -> look_backwards_for_SEMVAR_tag;
+		}
+		look_backwards_for_SEMVAR_tag -> bom_built
+    }
+
+
+	subgraph cluster_10 {
+		label = "Containers";
+		color=lightblue;
+
+		build_containers -> git_clone_from_BOM_hash_10;
+		subgraph cluster_11 {
+		    label = "For Each Service..."
+		    color=aquamarine3
+
+    		git_clone_from_BOM_hash_10 -> git_tag_SEMVAR_buildnum_10;
+    		git_tag_SEMVAR_buildnum_10 -> gcloud_container_build
+		}
+	}
+	gcloud_container_build -> publish_nightly_bom;
+
+	subgraph cluster_20 {
+		label = "Debians";
+		color=lightblue;
+
+		build_debians ->  git_clone_20;
+		subgraph cluster_21 {
+		    label = "For Each Service..."
+		    color=aquamarine3
+
+    		git_clone_20 -> checkout_hash_20;
+    		// delete tags
+    		checkout_hash_20 -> gradle_buildDeb_20;
+		}
+	}
+	gradle_buildDeb_20 -> publish_nightly_bom;
+
+
+	subgraph cluster_30 {
+		label = "RPMs";
+		color=lightblue;
+
+		build_rpms -> git_clone_from_BOM_hash_30;
+		subgraph cluster_31 {
+		    label = "For Each Service..."
+		    color=aquamarine3
+
+    		git_clone_from_BOM_hash_30 -> git_tag_SEMVAR_buildnum_30;
+    		// delete tags
+    		git_tag_SEMVAR_buildnum_30 -> gradle_buildRpm_30;
+
+		}
+	}
+	gradle_buildRpm_30 -> publish_nightly_bom;
+
+	subgraph cluster_40 {
+		label = "Changelog";
+
+		build_changelog -> git_clone_from_BOM_hash_40;
+		subgraph cluster_41 {
+		    label = "For Each Service..."
+		    color=aquamarine3
+
+    		git_clone_from_BOM_hash_40 -> look_backwards_for_SEMVAR_tag_commit;
+    		look_backwards_for_SEMVAR_tag_commit -> collect_commit_messages;
+
+		}
+		collect_commit_messages -> generate_markdown_to_file;
+	}
+	generate_markdown_to_file -> publish_nightly_bom;
+
+	subgraph cluster_50 {
+		label = "API Docs";
+
+		build_apidocs -> git_clone_gate_BOM_hash;
+		git_clone_gate_BOM_hash -> gradle_boot_run;
+		gradle_boot_run -> generate_swagger_docs_to_file;
+		generate_swagger_docs_to_file -> shutdown_gate;
+	}
+	shutdown_gate -> publish_nightly_bom;
+
+
+	subgraph cluster_60 {
+		label = "Publish";
+		color=lightgoldenrod
+
+		publish_nightly_bom -> git_clone_from_BOM_hash_60;
+
+        subgraph cluster_61 {git_clone_from_BOM_hash_60
+            label = "For Each Service..."
+	        color=aquamarine3
+
+	        git_clone_from_BOM_hash_60 -> get_hal_configs;
+	        get_hal_configs -> hal_admin_publish_profile;
+	        hal_admin_publish_profile -> hal_admin_publish_profile_alias_copy;
```

Comment:
I'm taking this copy out. Hal profiles are based on the component version, not the bom version. So this is in fact superfluous.